### ISSUE TITLE
S3 Access Denied - node_modules/aws-sdk/lib/services/s3.js

### ISSUE THREAD

**Issue by mithunkalan** [aws-amplify/amplify-js#2961]
**Describe the bug**
Unable to upload image to S3 using a FederatedSignIn from Google

**To Reproduce**
1. Get google token `const { type, accessToken, user } = await Google.logInAsync({ clientId });`
2. Signin into Cognito `Auth.federatedSignIn("google", { accessToken }, { name: user.id })`
3. Upload image   `Storage.put(text, Buffer.from(result.base64, "base64"), { contentType: "image/jpeg" })`
`[WARN] 16:35.678 AWSS3Provider - error uploading, [AccessDenied: Access Denied]`
`- node_modules/expo/build/environment/logging.js:25:23 in warn`
`- ... 15 more stack frames from framework internals`
`Access Denied`
`- node_modules/aws-sdk/lib/services/s3.js:580:34 in extractError`
`- node_modules/aws-sdk/dist/aws-sdk-core-react-native.js:3610:26 in callListeners`
`- ... 30 more stack frames from framework internals`

**Expected behavior**
I expected it to upload my image just like it does for Cognito users and Facebook users. 

**Extra info**
- I do exactly these steps with Facebook FederatedID signin & it works fine. Same with a cognito login. I looked at IAM roles & policies for the authorized user, nothing suggested that google is treated differently. If Facebook worked, so should Google.  
- I am able to read/write all AppSync tables with all users (cognito + FB + Goog). If AppSync works, why doesnt S3's Storage.put? They're working off the same role/policy.
- To get auth'd facebook to read/write AppSync & S3, I had to change `"aws_appsync_authenticationType": "AWS_IAM"` in my `aws-exports.js` and do the same for appsync's console. I also set the IAM role to have full read/write access to this AppSync's API + S3 bucket. 
- If the auth type was left as `COGNITO_USER_...?`, facebook & the unAuth role did not work with AppSync and facebook had "`no current user`" as an error. 

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.10
- Node v11.12.0, NPM 6.7.0, Expo-cli 2.13, amplify 1.1.8 
 - android 8.0.0 (Huawei p20lite)

**Comment by powerful23** [comment]
@mithunkalan yes there should be no difference between login from Fb or Google as long as both providers are configured correctly in the Cognito Federated Identity Pool. Can you do a research that when you federated sign in with Google, you will get an identity id. Can you search the identity id in the Cognito Federated Identity Pool -> Identity Browser to see if it's linked Google correctly.

**Comment by philiiiiiipp** [comment]
I am seeing the same issue, but with Cognito, after updating to the latest version.

After reverting back to:
```
"@aws-amplify/auth": "1.2.18",
"@aws-amplify/core": "1.0.23",
"@aws-amplify/storage": "1.0.26",
"@aws-amplify/cache": "1.0.23",
```

It works fine.

**Comment by mithunkalan** [comment]
@powerful23  @philiiiiiipp I think i got it fixed.
I investigated the identity browser. Saw that there were no google logins. Checked the response of Auth.federatedSignIn and saw that webIdentityCredentials.params.Logins."accounts.google.com" was always blank. This means that I was making an unauthenticated ID all the damn time. 
After a million combinations of ClientID , I find that I need to add the google API app's ID in the cognito federation ID pool section. All this time I thought it was the webClient ID that supposed to be there. Pretty sure the docs incorrectly tell me to use web.
